socket.go

```diff
@@ -38,6 +38,12 @@ import (
 
 type replyFunc func(err error, reply *replyOp, docNum int, docData []byte)
 
+type ScoketState int
```

Comment:
`SocketState` spelling

Reply:
You can also use a uint8 to (marginally) minimize the space needed for this type